backend/src/services/consumer-secret/consumer-secret-service.ts

```diff
@@ -0,0 +1,143 @@
+import { TPermissionServiceFactory } from "@app/ee/services/permission/permission-service";
+import { UnauthorizedError } from "@app/lib/errors";
+import { TConsumerSecretAttributeDALFactory } from "@app/services/consumer-secret-attribute/consumer-secret-attribute-dal";
+
+import { TKmsServiceFactory } from "../kms/kms-service";
+import { KmsDataKey } from "../kms/kms-types";
+import { TConsumerSecretDALFactory } from "./consumer-secret-dal";
+import {
+  TCreateConsumerSecretDTO,
+  TDeleteConsumerSecretDTO,
+  TGetConsumerSecretsDTO,
+  TUpdateConsumerSecretDTO
+} from "./consumer-secret-types";
+
+type TConsumerSecretServiceFactoryDep = {
+  consumerSecretDAL: TConsumerSecretDALFactory;
+  consumerSecretAttributeDAL: TConsumerSecretAttributeDALFactory;
+  kmsService: TKmsServiceFactory;
+  permissionService: Pick<TPermissionServiceFactory, "getUserOrgPermission">;
+};
+
+export type TConsumerSecretServiceFactory = ReturnType<typeof consumerSecretServiceFactory>;
+
+// todo: implement permissions
+export const consumerSecretServiceFactory = ({
+  consumerSecretDAL,
+  consumerSecretAttributeDAL,
+  kmsService
+}: TConsumerSecretServiceFactoryDep) => {
+  const createConsumerSecret = async ({
+    actorId,
+    actorOrgId,
+    actorAuthMethod,
+    attributes,
+    ...input
+  }: TCreateConsumerSecretDTO) => {
+    if (!actorOrgId) throw new UnauthorizedError({ message: "No organization ID provided in request" });
```

Comment:
Self-review: I think I should have checked that the user still has access to the organization

```
    const { permission } = await permissionService.getOrgPermission(actor, actorId, orgId, actorAuthMethod, actorOrgId);
    if (!permission) throw new ForbiddenRequestError({ name: "User does not belong to the specified organization" });
```